pkg/controller/controller.go

```diff
@@ -298,6 +313,21 @@ func (c *Controller) renderBadRequest(w http.ResponseWriter, m string) {
 	})
 }
 
+func getEmail(r *http.Request) (string, error) {
+	iapEmailValue := r.Header.Get(iapHeaderName)
+
+	if iapEmailValue == "" {
+		return "", fmt.Errorf("email header is not valid")
+	}
+
+	split := strings.Split(iapEmailValue, ":")
+	if len(split) != 2 {
+		return "", fmt.Errorf("email value has unexpected format, expected %s:<email>", iapHeaderName)
+	}
```

Comment:
https://cloud.google.com/iap/docs/signed-headers-howto#retrieving_the_user_identity says:

> Unlike that header and the `sub` claim, this value doesn't have a namespace prefix.

I don't think there's a colon split in this value.

Reply:
This is referring to the JWT post validation. We're actually using the other email header. Part of the issue I created addresses using the JWT from IAP.